### ISSUE TITLE
Optional limit

### ISSUE THREAD

**Issue by lgrahl** [juanluispaz/ts-sql-query#30]
Let's say I optionally want a `limit` clause.

```ts
let query = db
  .selectFrom(tMessage)
  .select(selectFields)
  .where(tMessage.conversationUid.equals(conversation))
  .orderBy('processedAt', 'desc');
if (limit !== undefined) {
  query = query.limit(limit);
}
return query.executeSelectMany();
```

That would become tedious quickly but it also doesn't work in TS because the type of `query` changes.

```ts
const query = db
  .selectFrom(tMessage)
  .select(selectFields)
  .where(tMessage.conversationUid.equals(conversation))
  .orderBy('processedAt', 'desc');
return limit !== undefined
  ? query.limit(limit).executeSelectMany()
  : query.executeSelectMany();
```

This works but I think we agree that it's awful, especially when there are more dynamic conditions.

```ts
return db
  .selectFrom(tMessage)
  .select(selectFields)
  .where(tMessage.conversationUid.equals(conversation))
  .orderBy('processedAt', 'desc')
  .limit(limit)
  .executeSelectMany();
```

This does not work but it would be my preferred solution. The `limit` method of the builder would just have to accept `null` or `undefined`.

Or maybe there are better ways I haven't found?

**Comment by juanluispaz** [comment]
Hi,

I always asked myself if this will be required; now you are requesting it 🤔

To deal with this situations I'm adding:
- `orderByFromStringIfValue`
- `limitIfValue` 
- `offsetIfValue`

This methods accepts null or undefined, and there is only applied when the provided argument have value.

Then, your query will be:

```ts
return db
  .selectFrom(tMessage)
  .select(selectFields)
  .where(tMessage.conversationUid.equals(conversation))
  .orderBy('processedAt', 'desc')
  .limitIfValue(limit)
  .executeSelectMany();
```

**Note**: It is not released yet

**Comment by juanluispaz** [comment]
Changes published in ts-sql-query 1.25.0
